test/rest_api/test_compaction_task.py

```diff
@@ -34,6 +36,7 @@ def check_compaction_task(cql, this_dc, rest_api, run_compaction, compaction_typ
                 for top_level_task in statuses:
                     if top_level_task["type"] != "resharding compaction":
                         check_child_parent_relationship(rest_api, top_level_task, depth)
+    drain_module_tasks(rest_api, module_name)
```

Comment:
I don't understand why we need to call this after the test after calling `wait_for_task` above.

Reply:
Because we wait only for tasks of given compaction type and during the test some compactions may be invoked in the background. 

It does not change anything for further tests (but memory) but these tasks will stay in the memory forever (what's set on purpose in this test case). Actually, it would make sense to drain all the modules in this case.

I am not sure if it's better to clear everything or leave it as is. What do you think?

Reply:
I think that the test should do whatever it needs to get a clean starting state
and cleanup after itself.
If the test would create tasks in other modules then it should clean those up
but if it only creates compaction tasks it should only clean the compaction module